packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -117,20 +162,46 @@ const ProductDetail = ({category, product, isLoading}) => {
             status: 'error'
         })
     }
+
+    const addItemToBasket = (basketId, variant, quantity) => {
+        console.log('basketId', basketId)
+        const productItems = [
+            {
+                productId: variant.productId,
+                quantity,
+                price: variant.price
+            }
+        ]
+
+        addItemToBasketAction.mutate(
+            {parameters: {basketId}, body: productItems},
+            {
+                onSuccess: () => {
+                    // only show this modal when a product is successfully add to cart
+                    onAddToCartModalOpen({product, quantity})
```

Comment:
Sometimes I would see an error when adding a product to cart. It seems the error happened to me during fresh cache.  Let me attach a video of it.

https://user-images.githubusercontent.com/847300/212991627-e66e8f45-1d03-4800-b52c-13d917acdf9c.mp4